sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BatchScanExec.scala

```diff
@@ -182,7 +182,16 @@ case class BatchScanExec(
 
               // Now fill missing partition keys with empty partitions
               val partitionMapping = nestGroupedPartitions.toMap
-              finalPartitions = spjParams.commonPartitionValues.get.flatMap {
+
+              // SPARK-41471: We keep to order of partition keys in `commonPartitionValues` to
```

Comment:
I think partition values should be sorted by themselves but just `mergePartitions` in `EnsureRequirements` doesn't maintain the ordering. 

Instead of doing it here, can we update `InternalRowComparableWrapper.mergePartitions` to make sure the result is ordered?